src/lib.rs

```diff
@@ -130,3 +159,111 @@ nonmax!(NonMaxU32, NonZeroU32, u32);
 nonmax!(NonMaxU64, NonZeroU64, u64);
 nonmax!(NonMaxU128, NonZeroU128, u128);
 nonmax!(NonMaxUsize, NonZeroUsize, usize);
+
+// https://doc.rust-lang.org/stable/src/core/convert/num.rs.html#383-407
+macro_rules! impl_nonmax_from {
+    ( $small: ty, $large: ty ) => {
+        impl From<$small> for $large {
+            #[inline]
+            fn from(small: $small) -> Self {
+                // SAFETY: smaller input type guarantees the value is non-zero
+                unsafe { Self::new_unchecked(small.get().into()) }
+            }
+        }
+    };
+}
+
+// Non-max Unsigned -> Non-max Unsigned
+impl_nonmax_from!(NonMaxU8, NonMaxU16);
+impl_nonmax_from!(NonMaxU8, NonMaxU32);
+impl_nonmax_from!(NonMaxU8, NonMaxU64);
+impl_nonmax_from!(NonMaxU8, NonMaxU128);
+impl_nonmax_from!(NonMaxU8, NonMaxUsize);
+impl_nonmax_from!(NonMaxU16, NonMaxU32);
+impl_nonmax_from!(NonMaxU16, NonMaxU64);
+impl_nonmax_from!(NonMaxU16, NonMaxU128);
+impl_nonmax_from!(NonMaxU16, NonMaxUsize);
+impl_nonmax_from!(NonMaxU32, NonMaxU64);
+impl_nonmax_from!(NonMaxU32, NonMaxU128);
+impl_nonmax_from!(NonMaxU64, NonMaxU128);
+
+// Non-max Signed -> Non-max Signed
+impl_nonmax_from!(NonMaxI8, NonMaxI16);
+impl_nonmax_from!(NonMaxI8, NonMaxI32);
+impl_nonmax_from!(NonMaxI8, NonMaxI64);
+impl_nonmax_from!(NonMaxI8, NonMaxI128);
+impl_nonmax_from!(NonMaxI8, NonMaxIsize);
+impl_nonmax_from!(NonMaxI16, NonMaxI32);
+impl_nonmax_from!(NonMaxI16, NonMaxI64);
+impl_nonmax_from!(NonMaxI16, NonMaxI128);
+impl_nonmax_from!(NonMaxI16, NonMaxIsize);
+impl_nonmax_from!(NonMaxI32, NonMaxI64);
+impl_nonmax_from!(NonMaxI32, NonMaxI128);
+impl_nonmax_from!(NonMaxI64, NonMaxI128);
+
+// Non-max Unsigned -> Non-max Signed
+impl_nonmax_from!(NonMaxU8, NonMaxI16);
+impl_nonmax_from!(NonMaxU8, NonMaxI32);
+impl_nonmax_from!(NonMaxU8, NonMaxI64);
+impl_nonmax_from!(NonMaxU8, NonMaxI128);
+impl_nonmax_from!(NonMaxU8, NonMaxIsize);
+impl_nonmax_from!(NonMaxU16, NonMaxI32);
+impl_nonmax_from!(NonMaxU16, NonMaxI64);
+impl_nonmax_from!(NonMaxU16, NonMaxI128);
+impl_nonmax_from!(NonMaxU32, NonMaxI64);
+impl_nonmax_from!(NonMaxU32, NonMaxI128);
+impl_nonmax_from!(NonMaxU64, NonMaxI128);
+
+// https://doc.rust-lang.org/stable/src/core/convert/num.rs.html#383-407
```

Comment:
```suggestion
// https://doc.rust-lang.org/1.47.0/src/core/convert/num.rs.html#383-407
```

Turning this into a permalink.